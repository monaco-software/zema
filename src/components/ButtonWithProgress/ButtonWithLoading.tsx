import React, { ComponentProps, FC } from 'react';
import { Button } from 'grommet';
import { Spinner, SpinnerProps } from '../Spinner/Spinner';

type ButtonProps = ComponentProps<typeof Button>;

interface Props extends ButtonProps {
  isLoading: boolean;
  spinnerSize?: SpinnerProps['size'];
  spinnerColor?: SpinnerProps['color'];
}

export const ButtonWithLoading: FC<Props> = ({ isLoading, spinnerSize, spinnerColor, ...restProps }) => {
  const spinnerIcon = isLoading
    ? <Spinner size={spinnerSize} color={spinnerColor} />
    : undefined;

  return (
    <Button
      disabled={isLoading}
      icon={spinnerIcon}
      {...restProps}
    />
  );
};

