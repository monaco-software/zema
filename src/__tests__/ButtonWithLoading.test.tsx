import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonWithLoading } from '@components/ButtonWithProgress/ButtonWithLoading';

it('<ButtonWithLoading /> should render plain button', () => {
  const tree = renderer
    .create(<ButtonWithLoading isLoading={false} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<ButtonWithLoading /> should render button with loading icon', () => {
  const tree = renderer.create(<ButtonWithLoading isLoading={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});
