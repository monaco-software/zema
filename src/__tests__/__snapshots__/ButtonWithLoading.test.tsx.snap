// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonWithLoading /> should render button with loading icon 1`] = `
<button
  className="StyledButton-sc-323bzc-0 iyUonS"
  disabled={true}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <svg
    fill="#7D4CDB"
    height={28}
    version="1.1"
    viewBox="0 0 32 32"
    width={28}
  >
    <path
      d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
      opacity=".25"
    />
    <path
      d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
    >
      <animateTransform
        attributeName="transform"
        dur="0.8s"
        from="0 16 16"
        repeatCount="indefinite"
        to="360 16 16"
        type="rotate"
      />
    </path>
  </svg>
</button>
`;

exports[`<ButtonWithLoading /> should render plain button 1`] = `
<button
  className="StyledButton-sc-323bzc-0 iZFCLM"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
/>
`;
